analytical_engine/core/fragment/dynamic_fragment.h

```diff
@@ -344,15 +347,23 @@ class DynamicFragment
       vid_t new_ivnum = vm_ptr_->GetInnerVertexSize(fid_);
       vid_t new_ovnum = ovgid_.size();
       is_selfloops_.resize(new_ivnum);
-      ie_.reserve_vertices(new_ivnum);
-      oe_.reserve_vertices(new_ivnum);
-      this->ivnum_ = new_ivnum;
+      this->ivnum_ = new_ivnum;      
       if (ovnum_ != new_ovnum) {
         ovnum_ = new_ovnum;
         initOuterVerticesOfFragment();
       }
-      if (!edges_to_add.empty()) {
-        addEdges(edges_to_add);
+      ie_.add_vertices(new_ivnum - ivnum_, new_ovnum - ovnum_);
+      oe_.add_vertices(new_ivnum - ivnum_, new_ovnum - ovnum_);
+      if (this->directed_) {
+        ie_.add_reversed_edges(edges_to_add);
+        oe_.add_forward_edges(edges_to_add);
+      } else {
+        ie_.add_edges(edges_to_add);
+        oe_.add_edges(edges_to_add);
```

Comment:
undirected graph only need to modify `oe_`

Reply:
Fixed